app/controllers/teacher_sets_controller.rb

```diff
@@ -59,32 +59,32 @@ def show
     @set = TeacherSet.find(params[:id])
 
     @active_hold = nil
-    is_ordered_max_quantity = false
+    is_ordered_max_quantity_ts = false
 
-    #Max copies value is configured(in elastic beanstalk). Based on this number teacher can order maximun number of sets.
+    #Max copies value is configured in elastic beanstalk.
+    #Max_copies_requestable is the maximum number of teachersets can request.
     max_copies_requestable = ENV['MAXIMUM_COPIES_REQUESTABLE'] || 5
 
     if @set.held_by? current_user
       @active_hold = @set.pending_holds_for_user(current_user).first
     end
 
-    #Calculates max ordered teacher set quantity from holds table.
-    ts_holds_count = @set.teacher_set_holds_count
-    is_ordered_max_quantity = true if ts_holds_count.to_i >= max_copies_requestable.to_i
+    #ts_holds_count is the number of holds currently held in the database for this teacher set.
+    ts_holds_count = @set.teacher_set_holds_count(current_user)
+
+    #is_ordered_max_quantity is a boolean value. Based on this flag disabled the order button in teacherset page.
+    #Button should be disabled after teacher has ordered maximum.
+    is_ordered_max_quantity_ts = true if ts_holds_count.to_i >= max_copies_requestable.to_i
 
     #Teacher set available copies less than configured value, we should show ts available_copies count in teacherset order dropdown.
-    if max_copies_requestable.to_i >= @set.available_copies.to_i && ts_holds_count.to_i <= 0
-      max_copies_requestable = @set.available_copies.to_i
-    elsif ts_holds_count.to_i >= 0
-      max_copies_requestable = [max_copies_requestable - ts_holds_count.to_i, @set.available_copies.to_i].min
-    end
+    max_copies_requestable = [max_copies_requestable - ts_holds_count.to_i, @set.available_copies.to_i].min
 
     render json: {
       :teacher_set => @set,
       :active_hold => @active_hold,
       :user => current_user,
       :allowed_quantities => (1..max_copies_requestable.to_i).to_a,
-      :is_ordered_max_quantity => is_ordered_max_quantity
+      :is_ordered_max_quantity_ts => is_ordered_max_quantity_ts
```

Comment:
One suggestion: Rather than compute `is_ordered_max_quantity_ts` in one way and then compute `max_copies_requestable` in a slightly different way, what if you computed `is_ordered_max_quantity_ts` *after* computing `max_copies_requestable` and base it off of that number? (i.e. `is_ordered_max_quantity_ts = max_copies_requestable > 0`).

Also, now that `is_ordered_max_quantity_ts` is scoped to the `user_id`, maybe call it something like `user_has_ordered_max`?

Alternatively, get rid of `is_ordered_max_quantity_ts` altogether and just set `:allowed_quantities` to `[]` when the user has hit their max. That seems semantically pretty clean to me.

Just suggestions with the aim to reduce the number of places the same idea is computed in slightly different ways